woocommerce/woocommerce-ios
# Deprecate `Deliverfile` in favor of `Fastfile`

## Description
Similarly to what was done for WPiOS on https://github.com/wordpress-mobile/WordPress-iOS/pull/16805, this PR removes the `Deliverfile` moving everything to a new lane in the `Fastfile`.
Additionally, the need for an `app_version` set in the `Deliverfile` has been deprecated and its usage is gonna soon be [removed altogether from release-toolkit](https://github.com/wordpress-mobile/release-toolkit/pull/447), so this PR already cleans this up.

## Testing instructions
Testing the lane is similar to https://github.com/wordpress-mobile/WordPress-iOS/pull/16805 (and is particularly difficult for me to do completely, so I'd also appreciate some help):
> 1. Checkout this branch
> 2. Run `bundle exec fastlane update_metadata_on_app_store_connect`
> 3. Verify the `Preview.html` contains valid data. In particular, that it's not attempting to upload screenshots
> 4. **Cancel the upload** 🙏 `Does the Preview on path './fastlane/Preview.html' look okay for you? (y/n) n`

Also good to test the use of the parameter `with_screenshots:true` to see that it will attempt to load the screenshots.

---
